tests/test_restriction_matrix.cc

```diff
@@ -230,6 +230,9 @@ class TestMeshEvaluator : public mfmg::DealIIMeshEvaluator<dim>
     constraints.clear();
     constraints.reinit(locally_relevant_dofs);
     dealii::DoFTools::make_hanging_node_constraints(dof_handler, constraints);
+    // The test requires us not to put boundary conditions
+    // dealii::VectorTools::interpolate_boundary_values(
+    //     dof_handler, 1, dealii::Functions::ZeroFunction<dim>(), constraints);
```

Comment:
Can you elaborate?  I don't really understand.

Reply:
The test assumes that the eigenvectors are constant. This is true for laplace if you don't have dirichlet bc. If you have dirichlet bc, the eigenvectors are not constant and the test fails.